src/fondant/pipeline/pipeline.py

```diff
@@ -319,11 +328,136 @@ def __init__(
+             cluster_type: t.Optional[str] = "default",
+             client_kwargs: t.Optional[dict] = None,
+             resources: t.Optional[Resources] = None,
+             schema: t.Optional[t.Dict[str, str]] = None) -> "Pipeline":
+
+        component_op = self._build_component_op(name,
+                                                arguments=arguments,
+                                                input_partition_rows=input_partition_rows,
+                                                cache=cache,
+                                                cluster_type=cluster_type,
+                                                client_kwargs=client_kwargs,
+                                                resources=resources,
+                                                schema=schema)
+
+        self.add_op(component_op)
+        return self
+
+
+    def apply(self,
+              name,
+              *,
+              arguments: t.Optional[t.Dict[str, t.Any]] = None,
+              input_partition_rows: t.Optional[t.Union[str, int]] = None,
+              cache: t.Optional[bool] = True,
+              cluster_type: t.Optional[str] = "default",
+              client_kwargs: t.Optional[dict] = None,
+              resources: t.Optional[Resources] = None,
+              consumes: t.Optional[t.Dict[str, str]] = None,
+              produces: t.Optional[t.Dict[str, str]] = None) -> "Pipeline":
+
+        component_op = self._build_component_op(name,
+                                                arguments=arguments,
+                                                input_partition_rows=input_partition_rows,
+                                                cache=cache,
+                                                cluster_type=cluster_type,
+                                                client_kwargs=client_kwargs,
+                                                resources=resources,
+                                                consumes=consumes,
+                                                produces=produces)
+
+        # Get previous component
+        previous_component = list(self._graph.items())[-1][1]["fondant_component_op"]
+        if previous_component is None:
+            msg = f"No previous component found."
+            raise ValueError(msg)
+
+        self.add_op(component_op, dependencies=previous_component)
+        return self
+
+    def write(self,
+              name,
+              *,
+              arguments: t.Optional[t.Dict[str, t.Any]] = None,
+              input_partition_rows: t.Optional[t.Union[str, int]] = None,
+              cache: t.Optional[bool] = True,
+              cluster_type: t.Optional[str] = "default",
+              client_kwargs: t.Optional[dict] = None,
+              resources: t.Optional[Resources] = None,
+              consumes: t.Optional[t.Dict[str, str]] = None,
+              schema: t.Optional[t.Dict[str, str]] = None) -> "Pipeline":
```

Comment:
Same comment as above, is the schema here just needed for renaming or does it also specify which fields to write?

Reply:
The schema should specify which fields and belonging types will be written.